### DISCUSSION TITLE
Custom In-Memory Reader

### DISCUSSION THREAD

**Discussion by crhoover-maxar** [cogeotiff/rio-tiler#674] [Ideas]
This is a feature request where it would be useful if there was a CustomReader that cached a file in-memory so that every tile request doesn't use I/O and could draw the content from memory instead of the filesystem

**Comment by vincentsarago** [comment]
@crhoover-maxar do you want to cache only the tiles or the whole file?

**Reply by AndrewAnnex** [reply]
I'd imagine both would be of interest, but I currently have a similar desire for the whole file
